snapcraft/internal/sources/errors.py

```diff
@@ -21,6 +21,26 @@ class VCSError(errors.SnapcraftError):
     fmt = '{message}'
 
 
+class SnapcraftSourceUnhandledError(errors.SnapcraftError):
+
+    fmt = ('Failed to pick an appropriate source handler:\n'
```

Comment:
These messages aren't ideal, they don't feel right

Reply:
Let's see. This is still hard for me, but what about:

Failed to get source: unknown source-type for {source}.
Run `snapcraft help sources` to read the documentation.

?

Reply:
Oh I want to join in!

I think Leo is on the right track here, but the issue isn't that the `source-type` is unknown (that would bite us [here](https://github.com/snapcore/snapcraft/pull/1929/files#diff-8c6bb564ac3a3ca26deb09b5c0b8b2e2R164) instead), it's that we're unable to _determine_ the source type given the source.

Remembering that our guidance is:

```
# <what>: What went wrong? Not "what technical detail broke in snapcraft" like is here now, but "what, that the user cares about, went wrong?"
# <why>: Why did that terrible thing happen?
# <fix>: How might the user carry on?
<what>: <why>
<fix>
```

(note that there is no line break between `<what>` and `<why>`. Something new we learned at the summit. @elopio double check me?)

If we can manage to get the part name into this, I suggest a combination of both Leo's suggestion and what's already there:

```
Failed to fetch source for part {part_name!r}: unable to determine source type of {source!r}.\n
Consider specifying `source-type`. See `snapcraft help sources` for more information.
```

Reply:
Thank you for these suggestions. I went with the last one, retaining the consideration for a typo in the URL.
Getting the part_name here is slightly painful since the source handlers are used in several different places... but I'm trying to figure it out, since actually it's really useful to have there.